virtual_rainforest/models/abiotic/energy_balance.py

```diff
@@ -0,0 +1,321 @@
+            self.mean_annual_temperature,
+            self.temperature_above_canopy,
+            self.canopy_layers + self.soil_layers,
+        )
+
+        self.air_temperature = temperature_interpolation[
+            (int(self.soil_layers)) : int((self.canopy_layers + self.soil_layers))
+        ]
+        self.soil_temperature = temperature_interpolation[0 : int(self.soil_layers)][
+            ::-1
+        ]
+        self.absorbed_radiation = absorbed_radiation
+        self.canopy_temperature = (
+            self.air_temperature + LEAF_TEMPERATURE_INI_FACTOR * self.absorbed_radiation
+        )
+
+        # initiate relative humidity and atmospheric pressure
+        self.relative_humidity = np.repeat(
+            data["relative_humidity_2m"], self.canopy_layers
+        )
+        self.atmospheric_pressure = data["atmospheric_pressure_2m"]
+
+        # set initial conductivities
+        self.air_conductivity = (
+            np.repeat(AIR_CONDUCTIVITY, self.canopy_layers + 1)
+            * (self.canopy_layers / self.canopy_height)
+            * (2 / self.canopy_layers)
+        )
+        self.air_conductivity[0] = 2 * self.air_conductivity[0]
+        self.air_conductivity[self.canopy_layers] = (
+            self.air_conductivity[self.canopy_layers]
+            * (self.canopy_height / self.canopy_layers)
+            * 0.5
+        )
+
+        self.leaf_conductivity = np.linspace(
+            np.array(MIN_LEAF_CONDUCTIVITY),
+            np.array(MAX_LEAF_CONDUCTIVITY),
+            self.canopy_layers,
+        )
+        self.air_leaf_conductivity = np.linspace(
+            np.array(MIN_LEAF_AIR_CONDUCTIVITY),
+            np.array(MAX_LEAF_AIR_CONDUCTIVITY),
+            self.canopy_layers,
+        )
+
+        # set initial heights
+        self.canopy_node_heights = [
+            (x + 0.5) / self.canopy_layers * self.canopy_height
+            for x in range(0, self.canopy_layers)
+        ]
+        self.soil_node_depths = [
+            self.soil_depth
+            / (self.soil_layers**SOIL_DIVISION_FACTOR)
+            * (x**SOIL_DIVISION_FACTOR)
+            for x in range(1, self.soil_layers + 1)
+        ]
+        self.height_of_above_canopy = self.canopy_height
+        self.canopy_wind_speed = [
+            (x / self.canopy_layers) * self.wind_speed_10m
+            for x in range(1, self.canopy_layers + 1)
+        ]
+
+        # set initial fluxes
+        self.sensible_heat_flux = np.array(0.0)
+        self.latent_heat_flux = np.repeat(0.0, self.canopy_layers)
+        self.ground_heat_flux = np.array(0.0)
+        self.diabatic_correction_factor = np.array(0.0)
+
+        # get soil parameters
+        self.soil_parameters = soil_parameters[str(soil_type)]
+
+    def calc_molar_density_air(
+        self,
+        temperature: NDArray[np.float32] = np.array(20.0, dtype=float),
+        atmospheric_pressure: NDArray[np.float32] = np.array(101.3, dtype=float),
```

Comment:
If you use `dtype=float`, I think that's the same as `np.float64`, which is not what you've put in the type hint. I'd just use `np.float32` everywhere.